app/controllers/krikri/providers_controller.rb

```diff
@@ -0,0 +1,19 @@
+module Krikri
+  # This models a provider that is stored in Marmotta.
+  class ProvidersController < ApplicationController
+    def index
+      @providers = Krikri::Provider.all
+    end
+
+    def show
+      if params[:set_session]
+        session[:current_provider] = params[:id]
+        redirect_to :back, provider: params[:id]
+      elsif params[:clear_session]
+        session.delete :current_provider
+        redirect_to providers_path
```

Comment:
This *is* working as expected, but isn't really the desired behavior.  It redirects session variable clears (clicks on `All Providers`) to the providers dashboard, rather than the provider-less version of the calling page.